msm/src/column_env.rs

```diff
@@ -91,12 +92,33 @@ impl<'a, const N: usize, F: FftField> TColumnEnvironment<'a, F> for ColumnEnviro
         }
     }
 
-    fn column_domain(&self, _col: &Self::Column) -> Domain {
+    fn column_domain(&self, col: &Self::Column) -> Domain {
         // TODO FIXME check this is a tricky variable it should match the evalution in column
         // this must be bigger or equal than degree chosen in runtime inside evaluations() for
         // evaluating an expression = degree of expression that is evaluated
         // And also ... in some cases... bigger than the witness column size? Equal?
-        Domain::D4
+        match *col {
+            Self::Column::X(i) => {
+                let domain_size = self.witness[i].domain().size;
+                if self.domain.d1.size == domain_size {
+                    Domain::D1
+                } else if self.domain.d2.size == domain_size {
+                    Domain::D2
+                } else if self.domain.d4.size == domain_size {
+                    Domain::D4
+                } else if self.domain.d8.size == domain_size {
+                    Domain::D8
+                } else {
+                    panic!("Domain not supported. We do support the following multiple of the domain registered in the environment: 1, 2, 4, 8")
+                }
+            }
+            Self::Column::LookupAggregation
```

Comment:
This is implemented later, in https://github.com/o1-labs/proof-systems/pull/1999.